lib/amazon-pricing/common/ec2_common.rb

```diff
@@ -41,14 +41,38 @@ def fetch_ec2_instance_pricing_ri_v2(url, operating_system)
           next
         end
         reg['instanceTypes'].each do |type|
-          api_name = type["type"]
-          instance_type = region.get_instance_type(api_name)
+          # it's possible we're missing the instance_type (i.e. it wasn't in the base, cf: ec2-di-price-list),
```

Comment:
can I suggest a more readable organization of code:

reg['instanceTypes'].each do |type|
 ensure_existence_of_instance_type type
  ...
end

def ensure_existence_of_instance_type type
  # Create instance type
  instance_type = find_or_create_instance_type type
  # Fill in od price if missing
  fill_od_price_if_missing instance_type
end

def find_or_create_instance_type type
...
end

def fill_od_price_if_missing instance_type
...
end